### DISCUSSION TITLE
4.0.0+'s binary size is larger than 3.00's

### DISCUSSION THREAD

**Discussion by GaryElshaw** [transmission/transmission#2840] [General]
Bauhaus - The Sky's Gone Out Pt. 2

“My baby. How big you’ll be in a very little while. You’ll be going to school and you won’t want your daddy then, will you darling?" 

![SCR-20220330-mra](https://user-images.githubusercontent.com/69029666/160745219-85e10b3f-59c8-4ce7-bdb9-c757eb4588a7.png)

"Oh, I wish you could be my baby forever. I wonder what the future holds?”

![SCR-20220330-ms2](https://user-images.githubusercontent.com/69029666/160745252-c1ae8142-255d-4d47-bdf7-7c45f3d22822.png)

**Comment by sweetppro** [comment]
Fat binary.
X64 + arm = 2x the size

**Reply by GaryElshaw** [reply]
Not to be fattest, but... :-)

Actually, it's been recent, I think it was the FMT changes.

**Comment by ckerr** [comment]
You're comparing a release build to a nightly debug build though?

*edit 1*: [As previously discussed](https://github.com/transmission/transmission/pull/3351#issuecomment-1168123675) *please* make description titles comprehensible. If the topic is "the binary size is larger in 4 than in 3" then someone searching for that is *probably* not going to use "Bauhaus" as the search key just because of one of their lyrics.

*edit 2*: Also, even a release stripped binary w/o debug symbols is probably somewhat larger in 4 since it's written in C++ rather than C but the project is not going back to straight C :smile_cat:

**Comment by pmarsden** [comment]
Came across this in a search today.

My 3.00 built on Centos 7, stripped = 534KB
A test 4.00 built on Centos 8, stripped = 1,547KB

An almost 3x increase - is this correct ?
I thought it was a build fault at the time.

[ Yes, this was a few months ago ].

**Reply by ckerr** [reply]
Might be correct? Transmission 4 is a massive rewrite relative to Transmission 3.

It's still smaller than `rtorrent 0.9.8` (2,072 KB stripped) and `qBittorrent v4.4.1` (9,704 KB stripped) on my machine (Ubuntu 22.04.2) so :man_shrugging:

**Reply by pmarsden** [reply]
I'm not concerned about the specific size as such.
It was the massive increase in size that made me think there was an issue.
I didnt proceed to look into it any further at the time (I'm only running C8 on a test box).

**Comment by fulalas** [comment]
The new version is not just larger, it's WAY larger.

`transmission-gtk` binary:
v3.00: 876.9 KiB
v4.03: 3640.9 KiB

This is a huge regression in my opinion.

**Reply by ckerr** [reply]
Patches welcomed
